### ISSUE TITLE
Clarify/correct iOS Safari entry

### ISSUE THREAD

**Issue by ssokolow** [MaulingMonkey/cargo-html#43]
The browser support table in the README lists `iOS Safari | ✔️ 14.4+` (copy-pasted from lib.rs because the emoji doesn't copy-paste from the GitHub rendering) but, in my tests, (using a minimal HTML file that just uses a control string, a `<noscript>` element, and ` document.write`), iOS 14.4.2 Safari seems to insist on running with JavaScript disabled when loading an HTML file off the internal storage via the Files app.

(I think it's just insisting on using the embeddable WebKit view without a means to load local files in full-blown Safari and, in typical iOS fashion, using "simple" UIs to make things ambiguous and confusing.)

This makes the README either very misleading (if the single-file HTML is only useful on iOS when served from a remote server), or in need of clarification on how to enable the supported configuration.

Even if JavaScript *is* unavailable on iOS Safari, I'd still be interested in using cargo-html if HTML+CSS without JavaScript or WebAssembly support is a supported configuration, since it'd mean I could retire the Python-based analogue to cargo-html that I hacked together for translating CSV-format "books to watch for a the used book store" lists into expandable/collapsible/filterable HTML outlines to be copied onto a phone with no SIM. (The output of which does work on Firefox for Android, pre-"you get the six extensions we like and you'll be happy with it" version but not in iOS Safari.)

**Comment by MaulingMonkey** [comment]
> I think it's just insisting on using the embeddable WebKit view without a means to load local files in full-blown Safari and, in typical iOS fashion, using "simple" UIs to make things ambiguous and confusing.

Agreed - Files is probably directly previewing the HTML via WebKit from your description of your workflow.

WebKit can at least theoretically run JavaScript even for locally stored files, based on https://stackoverflow.com/questions/53133012/loading-local-html-with-local-javascript-sources-in-a-wkwebview-ios-swift , but it's possible that JavaScript is outright disabled if `allowingReadAccessTo` isn't provided a directory.  There's also properties on `WKPreferences` such as `javaScriptEnabled` (deprecated, but possibly still working?)  Not sure what the default of those are, or what apps might muck with said defaults.  It might be worth trying other alternative file/browser/reader apps that use WebKit, to see if any of them will work.

Sadly, I currently lack access to modern iOS hardware for investigating this and testing.

> This makes the README either very misleading (if the single-file HTML is only useful on iOS when served from a remote server), or in need of clarification on how to enable the supported configuration.

A big 'ole `iOS Files | ❌ WebKit not configured to run JavaScript` might be appropriate immediately below `iOS Safari | ...`?  I'd also be interested in what apps (if any) I could recommend in the Readme as alternatives.

> I'd still be interested in using cargo-html if HTML+CSS without JavaScript or WebAssembly support is a supported configuration

There are static site generators written in Rust which might be useful for this route, but cargo-html's entire focus is on generating/gluing together JavaScript and WebAssembly for dynamic web apps.

While you could theoretically provide a custom template that just ignores the JavaScript and WebAssembly bits, at that point cargo-html would be just a massively overcomplicated file copy command, and wouldn't really be contributing anything.

**Comment by MaulingMonkey** [comment]
> 32a4cf85e2aa480d94ebfeb05ea38f35bd46c5d1 Readme.md: update iOS portability

Aside from adding an `iOS Files` entry, I've also added caveats in the `iOS Safari` entry:

| Browser               | Supported Version |
| -----------------     | ----------------- |
| ...                         | ...
| iOS Safari            | ✔️ 14.4+ (webserver required, can't load local files)
| iOS Files             | ❌ None? (JavaScript disabled? [#43](https://github.com/MaulingMonkey/cargo-html/issues/43))
| ...                         | ...

Please let me know if you think this is still misleading, or if you find another iOS app which allows JavaScript execution for local files.

**Comment by ssokolow** [comment]
>  I'd also be interested in what apps (if any) I could recommend in the Readme as alternatives.

Unfortunately, I can't help with the other app suggestions.

My only iDevice is a hand-me-down that I use for testing my web creations and I'm basically just using it as a fixed-function Safari+Files+Camera device because I'm ideologically opposed to Apple's app-signing model.
